### DISCUSSION TITLE
Significance of Class DiffEqLayer

### DISCUSSION THREAD

**Discussion by kapilVetrivel** [wkirgsn/thermal-nn#8] [General]
Hello!

I'm trying to understand why the Class DiffEqLayer is present & how `DiffEqLayer(TNNCell)` works.
Is there any resources/examples to understand the same ?

Thanks

**Comment by wkirgsn** [comment]
Hi Kapil,
while the TNNCell declares how a single observation is handled, the DiffEqLayer wrapper declares how a sequence of observations is to be fed into the cell class. This way you can swap different cells without the need to implement boilerplate code.

A similar design pattern can be found in Pytorch with [GRUCell and LSTMCell](https://pytorch.org/docs/stable/generated/torch.nn.LSTMCell.html#torch.nn.LSTMCell) classes
